libs/ui-lib/lib/ocm/components/featureSupportLevels/FeatureSupportLevelProvider.tsx

```diff
@@ -123,80 +70,81 @@ export const FeatureSupportLevelProvider: React.FC<SupportLevelProviderProps> =
     [cluster?.cpuArchitecture, infraEnv?.cpuArchitecture, infraEnv?.staticNetworkConfig],
   );
 
-  const getVersionSupportLevelsMapCallback = React.useCallback(
-    (versionName: string): FeatureIdToSupportLevel | undefined => {
-      return getVersionSupportLevelsMap(versionName, supportLevelData);
-    },
-    [supportLevelData],
-  );
+  const getFeatureSupportLevels = React.useCallback((): NewFeatureSupportLevelMap => {
+    return supportLevelData;
+  }, [supportLevelData]);
 
   const getFeatureSupportLevel = React.useCallback(
-    (versionName: string, featureId: FeatureId): SupportLevel | undefined => {
-      const versionSupportLevelData = getVersionSupportLevelsMapCallback(versionName);
-      return versionSupportLevelData ? versionSupportLevelData[featureId] : undefined;
+    (
+      featureId: FeatureId,
+      supportLevelDataNew?: NewFeatureSupportLevelMap,
+    ): SupportLevel | undefined => {
       // eslint-disable-next-line react-hooks/exhaustive-deps
+      if (supportLevelDataNew) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        return supportLevelDataNew ? (supportLevelDataNew[featureId] as SupportLevel) : undefined;
+      } else {
+        if (supportLevelData) {
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          return supportLevelData ? (supportLevelData[featureId] as SupportLevel) : undefined;
+        }
+      }
     },
-    [getVersionSupportLevelsMapCallback],
+    [supportLevelData],
   );
 
   const isFeatureSupportedCallback = React.useCallback(
-    (versionName: string, featureId: FeatureId) => {
-      const supportLevel = getFeatureSupportLevel(versionName, featureId);
-      return isFeatureSupported(versionName, featureId, supportLevel, versionOptions);
+    (featureId: FeatureId, supportLevelDataNew?: NewFeatureSupportLevelMap) => {
+      const supportLevel = getFeatureSupportLevel(featureId, supportLevelDataNew);
+      return isFeatureSupportedAndAvailable(supportLevel);
     },
-    [getFeatureSupportLevel, versionOptions],
+    [getFeatureSupportLevel],
   );
 
-  //TODO(brotman): move to separate context FeatureStateContext
   const getDisabledReasonCallback = React.useCallback(
-    (versionName: string, featureId: FeatureId) => {
-      const isSupported = isFeatureSupportedCallback(versionName, featureId);
-      return getFeatureDisabledReason(
+    (featureId: FeatureId, t?: TFunction, supportLevelDataNew?: NewFeatureSupportLevelMap) => {
```

Comment:
Why do we need to add the `t` function?
Also, for the `supportLevelDataNew`, could we not directly use `featureSupportLevels`?

It looks like we're not taking advantage of the fact that this interface is now separate from the CIM side, and we're doing many unneeded parts that make the code much more complex.

Reply:
True. I've changed it